### ISSUE TITLE
[user]. listing. divergence

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#477]
v1.22.61
```
?- [user].
a.
^D
   true.

?- listing.
a.
   true.

?- [user].
a.
Warning: overwriting a/0
.
Error: syntax error, incomplete statement, line 2 '.
'
   true.

?- listing.
   true.

?- [user].
a.
Warning: overwriting a/0   % which one? listing does not list any a.

```

**Comment by flexoron** [comment]
Somehow swapped. Now missing overwriting message
v1.23.39
```
$ ./tpl
?- [user].
a.
   true.
?- listing.
a.
   true.

?- [user].
a:-true.
Warning: overwriting a/0
   true.
?- listing.
a :- true.
   true.

?- [user].
a:-false.
          % warning message absent
   true.
?- listing.
a :- false.
   true.
```

**Comment by flexoron** [comment]
swapped again. 
v1.23.40
```
$ ./tpl
?- [user].
a.
   true.
?- [user].
a.
Warning: overwriting a/0
.
Error: syntax error, incomplete statement, line 2 '.
'
   true.

?- listing.
   true.

?- [user].
a.
Warning: overwriting a/0
   true.
```

**Comment by flexoron** [comment]
fixed with v1.23.42
